### PR TITLE
Convert DWA and GP into a MBF plugin with specific error codes and messages

### PR DESCRIPTION
**Pull request by corot**
https://www.wrike.com/open.htm?id=924956015
https://www.wrike.com/open.htm?id=924956102

btw, should we rename the target branch to `devel`?

### CODE REVIEW

File: dwa_local_planner/src/dwa_planner_ros.cpp [hunk 1]
```diff
@@ -263,28 +268,30 @@ namespace dwa_local_planner {
...
+      message = "Could not get robot pose";
+      ROS_ERROR_STREAM_NAMED("dwa_local_planner", message);
+      return mbf_msgs::ExePathResult::TF_ERROR;
     }
     std::vector<geometry_msgs::PoseStamped> transformed_plan;
     if ( ! planner_util_.getLocalPlan(current_pose_, transformed_plan)) {
-      ROS_ERROR("Could not get local plan");
-      return false;
+      message = "Could not get local plan";
+      ROS_ERROR_STREAM_NAMED("dwa_local_planner", message);
```
Comment (renan028) [inline dwa_local_planner/src/dwa_planner_ros.cpp:286]: S: create a static member for this `dwa_local_planner`

**Review by siferati — APPROVED** [review]
I'm in favor of renaming branch to devel.

**Review by renan028 — APPROVED** [review]

File: dwa_local_planner/src/dwa_planner_ros.cpp [hunk 2]
```diff
@@ -157,7 +159,7 @@ namespace dwa_local_planner {
     return dp_->setPlan(orig_global_plan);
   }
 
-  bool DWAPlannerROS::isGoalReached() {
+  bool DWAPlannerROS::isGoalReached(double xy_tolerance, double yaw_tolerance) {
```
Comment (ChristofDubs) [inline dwa_local_planner/src/dwa_planner_ros.cpp]: if the arguments are unused, you could just omit the names, i.e. `isGoalReached(double, double)`

**Review by ChristofDubs — APPROVED** [review]

File: dwa_local_planner/src/dwa_planner_ros.cpp (hunk 1 above)
Comment (corot) [reply dwa_local_planner/src/dwa_planner_ros.cpp:286]: oos, and anyway we won't use this for long

**Pull request merged**